aesara/tensor/math_opt.py

```diff
@@ -3495,6 +3504,79 @@ def local_reciprocal_1_plus_exp(fgraph, node):
                         return out
 
 
+# x_lv, if any, is the logic variable version of some term x,
+# while x_at, if any, is the Aesara tensor version for the same.
+def distribute_over_add(in_lv, out_lv):
+    from cons import cons
+    from kanren import eq, fact, heado, tailo
+    from kanren.assoccomm import eq_assoccomm
+
+    # This does the optimization
+    # 1. (x + y + z) * A  = x * A + y * A + z * A
+    # 2. (x + y + z) / A  = x / A + y / A + z / A
```

Comment:
Could this lead to numerical issues?

Reply:
Ah no, the actual optimization being done here is reverse of that:

```
# Line 3545
distribute_over_add_opt = KanrenRelationSub(lambda x, y: distribute_over_add(y, x))
```

So it combines the additive terms not distributes it. It's just that it's easier to represent it in this way in `kanren` and it works just as well if we implement it the other way round purely in `kanren`.